### PR TITLE
Add editorconfig, recommended extensions

### PR DESCRIPTION
**Pull request by nonowazu**
**Purpose**
In an effort to improve dev experience with nari, I'm adding a .editorconfig file and suggest some workspace extensions (python support, pylance, and editorconfig support)

**New Features**
Improved dev experience if they opt into the recommended extensions

**Bug Fixes**
This might fix a bug where someone cut/pastes code from me and I type it badly and it has wrong indents.

**Before/After**
N/A

**Additional Context**
Python 'class inheritance' is actually an expression. This is valid python:

```python
try:
    from foo import Foo
except ImportError:
    Foo = None
    from bar import Bar

class SomeClass(Foo or Bar):
    pass
```

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+[*]
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+charset = utf-8
+
+# 4 space indentation
+[*.py]
+indent_style = space
+indent_size = 4
```
Comment (ayyaruq) [inline .editorconfig:14]: Can we make it 2 space and tab or will that shit up pylint?

File: .vscode/extensions.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+	"recommendations": [
+        "ms-python.python",
+        "ms-python.vscode-pylance",
+        "EditorConfig.EditorConfig"
```
Comment (ayyaruq) [inline .vscode/extensions.json:5]: I thought this one was all lowercase. YOLO tho.

**Review by ayyaruq — COMMENTED** [review]
LGTM but I think 4 space indents, that are actual spaces, is kinda awkward and only pylint likes it. Thoughts?

File: .editorconfig (hunk 1 above)
Comment (nonowazu) [reply .editorconfig:14]: 4 spaces is the [pep8](https://peps.python.org/pep-0008/#indentation) standard. If we switch to another linter, I don't want to configure it for a spacing that nobody else will obey.

File: .vscode/extensions.json (hunk 2 above)
Comment (nonowazu) [reply .vscode/extensions.json:5]: I right-clicked -> copy extension ID:

Screen Shot 2022-04-25 at 8 09 50 PM

So unless vscode messes up copying, it's probably good?

**Review by ayyaruq — APPROVED** [review]

**Pull request merged**